### ISSUE TITLE
Consistent handling of imports from files with single/multiple queries

### ISSUE THREAD

**Issue by Cito** [apollographql/graphql-tag#137]
PR #122 provided multiple query definitions per query file which is great.

However, there is currently a case distinction between files with a single query where the query must be imported directly, and files with multiple queries which must be always imported by name. You can neither import a single query by name, nor the first query directly as default.

This creates a problem when refactoring graphql files, adding a query to a file that so far held only a single query, or removing queries so that only one is left. In both cases, existing import statements break, often resulting in hard to debug error messages at run time. See also the comments in PR #122.

A solution might be to export single queries by name and as default value.

**Comment by rosskevin** [comment]
I'm also wondering about consistent usage in typescript:

```js
import * as GQL from './my.gql'

GQL.MyQuery
GQL.MyMutation
```

Sometimes I have multiple in a file, sometimes I have one, but I want to use the same pattern throughout the codebase and this issue prevents it.

**Comment by b1f6c1c4** [comment]
Yeah, I think it's a bug that impairs working experience.

A possible workaround is to append the following code to graphql file:
```graphql
# Hack graphql-tag/loader
# [bug](https://github.com/apollographql/graphql-tag/issues/137)
query Hack { hack }
```
It'll force a one-query graphql to become multi-query.

**Comment by detrohutt** [comment]
If you guys are using babel you could use the beta version (`2.2.0-beta.1`) of [my plugin](https://github.com/detrohutt/babel-plugin-inline-import-graphql-ast) in place of the webpack loader. It has exactly the behavior @Cito described.

If you try it out, let me know how it works for you so I can feel better about pushing to master. :)

**Comment by Cito** [comment]
Unfortunately, I'm using TypeScript. But thanks for pointing that out, maybe your plugin can be used as a reference implementation.

**Comment by jnwng** [comment]
this seems pretty reasonable to me and shouldn't be a breaking change if we export both a default and a named export for queries with a single operation. 

willing to accept PR's for this! otherwise hopefully can give it a try this weekend.

**Comment by jnwng** [comment]
closed by #154 and released in `graphql-tag@2.8.0`
